classes/services/NavigationMenuService.inc.php

```diff
@@ -83,6 +134,124 @@ function getDisplayStatusCallback($hookName, $args) {
 						null
 					));
 					break;
+				case NMI_TYPE_NEW_RELEASE:
+					$navigationMenuItem->setUrl($dispatcher->url(
+						$request,
+						ROUTE_PAGE,
+						null,
+						'catalog',
+						'newReleases',
+						null
+					));
+					break;
+				case NMI_TYPE_SERIES:
+					if ($relatedObject) {
+						$navigationMenuItem->setUrl($dispatcher->url(
+							$request,
+							ROUTE_PAGE,
+							null,
+							'catalog',
+							'series',
+							$relatedObject->getPath()
+						));
+					} else {
+						$navigationMenuItem->setIsDisplayed(false);
+					}
+					break;
+				case NMI_TYPE_CATEGORY:
+					if ($relatedObject) {
+						$navigationMenuItem->setUrl($dispatcher->url(
+							$request,
+							ROUTE_PAGE,
+							null,
+							'catalog',
+							'category',
+							$relatedObject->getPath()
+						));
+					} else {
+						$navigationMenuItem->setIsDisplayed(false);
+					}
+					break;
+			}
+		}
+	}
+
+	function getFormTemplateParametersCallback($hookName, $args) {
+		$templateParameters =& $args[0];
+
+		$request = \Application::getRequest();
+		$context = $request->getContext();
+		$contextId = $context ? $context->getId() : CONTEXT_ID_NONE;
+
+		$seriesDao = \DAORegistry::getDAO('SeriesDAO');
+		$series = $seriesDao->getByContextId($contextId);
+		$seriesTitlesArray = $series->toAssociativeArray();
+
+		$seriesTitles = array();
+		foreach ($seriesTitlesArray as $series) {
+			$seriesTitles[$series->getId()] = $series->getLocalizedTitle();
+		}
+
+		$categoryDao = \DAORegistry::getDAO('CategoryDAO');
+		$categories = $categoryDao->getByParentId(null, $contextId);
+		$categoryTitlesArray = $categories->toAssociativeArray();
+
+		$categoryTitles = array();
+		foreach ($categoryTitlesArray as $category) {
+			$categoryTitles[$category->getId()] = $category->getLocalizedTitle();
+		}
+
+		$ompTemplateParameters = array(
+			'navigationMenuItemSeriesTitles' => $seriesTitles,
+			'navigationMenuItemCategoryTitles' => $categoryTitles,
+		);
+
+		$templateParameters = array_merge($templateParameters, $ompTemplateParameters);
+	}
+
+	function getFormExecuteCallback($hookName, $args) {
+		$form =& $args[0];
+		$navigationMenuItem =& $args[1];
+
+		if ($form->getData('menuItemType') == NMI_TYPE_SERIES) {
+			$navigationMenuItem->setPath($form->getData('relatedSeriesId'));
+		} else if ($form->getData('menuItemType') == NMI_TYPE_CATEGORY) {
+			$navigationMenuItem->setPath($form->getData('relatedCategoryId'));
```

Comment:
Just nitpicking, but what do you think about using the `assocId` here instead of `relatedSeriesId` and `relatedCategoryId`? It's a pattern we already use heavily and you could get remove some of the conditional logic around where to find the associated object ID.

Reply:
@NateWr if I use assocId i think that there will be 2 id=assocId in the form. Won't that cause conflicts on form execute?

Reply:
You're right, leave it as is.